api/src/transactions.rs

```diff
@@ -1230,7 +1231,18 @@ impl TransactionsApi {
                 let mut user_transactions = Vec::new();
                 for transaction in transactions.into_iter() {
                     match transaction {
-                        Transaction::UserTransaction(user_txn) => user_transactions.push(*user_txn),
+                        Transaction::UserTransaction(user_txn) => {
+                            let mut txn = *user_txn;
+                            match vm_status {
+                                VMStatus::Error { .. } | VMStatus::ExecutionFailure { .. } => {
+                                    txn.info.vm_status +=
```

Comment:
Do we want to print all the glory of the VM status here? There is already a lot of info produced in [`explain_vm_status`](https://github.com/aptos-labs/aptos-core/blob/edcae567b88d706b848e89885005efc772259715/api/types/src/convert.rs#L965). Perhaps we only want to append the message here.

Reply:
I thought the location info might be helpful as well?

Reply:
I'm just concerned that the output looks confusing if the same info is there multiple times. Location info absolutely important, but check out the function I mention, it already prints some location info, even makes it prettier.

Reply:
And BTW, from my code search it appears `info.vm_status` is populated from the explain function.